### ISSUE TITLE
Scroll event continues to fire after destroy is called

### ISSUE THREAD

**Issue by swixpop** [metafizzy/flickity#482]
If I call flkty.destroy() while a slide transition is happening, the scroll event continues to fire. Other events are possibly firing as well. The result is that not all styles are being cleared from slides after destroy is called.

**Comment by desandro** [comment]
Thanks for reporting this bug. I'll have to take a look.

**Comment by jbach** [comment]
This can be reproduced via
```js
flkty.next();
flkty.destroy()
```
Calling `flkty._events = {};` after `flkty.destroy();` stops events from being called, but this not enough, as the animation loop keeps going, even if I set `flkty.isAnimating = false;`.

Maybe adding
```js
if ( !this.isActive ) {
  return;
}
```
to [`proto.animate`](https://github.com/metafizzy/flickity/blob/8c0ef69575ac1c24271068a8d71e6f781955d4f2/js/animate.js#L63) would help stopping the animation immediately?

**Comment by lebowski89** [comment]
Has there been any progress on this?

**Comment by desandro** [comment]
@jbach Good call. I'll be adding a similar fix in.

**Comment by desandro** [comment]
This fix has been released with [v2.2.0](https://github.com/metafizzy/flickity/releases/tag/v2.2.0). Thanks again!
